src/libraries/System.Security.Cryptography/tests/X509Certificates/PfxIterationCountTests.CustomAppDomainDataLimit.cs

```diff
@@ -0,0 +1,139 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.DotNet.RemoteExecutor;
+using Microsoft.DotNet.XUnitExtensions;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using Xunit;
+
+namespace System.Security.Cryptography.X509Certificates.Tests
+{
+    // AppContext and AppDomain are the same in this context.
+    public class PfxIterationCountTests_CustomAppDomainDataLimit
+    {
+        // We need to use virtual in a non-abstract class because RemoteExecutor can't work on abstract classes.
+        internal virtual X509Certificate Import(byte[] blob) => new X509Certificate(blob);
+
+        [ConditionalTheory(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        [MemberData(memberName: nameof(PfxIterationCountTests.GetCertsWith_IterationCountNotExceedingDefaultLimit_AndNullOrEmptyPassword_MemberData), MemberType = typeof(PfxIterationCountTests))]
+        public void Import_AppDomainDataWithValueTwo_ActsAsDefaultLimit_IterationCountNotExceedingDefaultLimit(string name, bool usesPbes2, byte[] blob, long iterationCount)
+        {
+            _ = iterationCount;
+            _ = blob;
+
+            if (usesPbes2 && !PfxTests.Pkcs12PBES2Supported)
+            {
+                throw new SkipTestException(name + " uses PBES2 which is not supported on this version.");
+            }
+
+            RemoteExecutor.Invoke((certName) =>
+            {
+                AppDomain.CurrentDomain.SetData("System.Security.Cryptography.Pkcs12UnspecifiedPasswordIterationLimit", -2);
```

Comment:
The runtime code reads data from `AppContext`, not from `AppDomain`. Consider using `AppContext.SetData` instead of `AppDomain.SetData` for all these tests since that's a closer representation of what code paths we're intending to exercise.

Reply:
Sure I can use AppContext.SetData for 8, I did it this way for 6 (and 7) because that API was introduced on 7, and I found this comment https://github.com/dotnet/runtime/issues/47922#issuecomment-869012343 suggesting that AppDomain.SetData gives the same result.